### ISSUE TITLE
Using ltex-ls on markdown and raw cells

### ISSUE THREAD

**Issue by teucer** [jupyter-lsp/jupyterlab-lsp#865]
I want to use [ltex-ls](https://github.com/valentjn/ltex-ls) for markdown and raw cells. Are there any pointers on how to achieve that?

**Comment by krassowski** [comment]
Yes, this is in scope but not currently possible. This was discussed previously in #34. I did not know about 
ltex-ls before and was waiting for a revival of non-maintained [languagetool-languageserver](https://github.com/languagetool-language-server/languagetool-languageserver) before putting more time into this, but `ltex-ls` seems like an even better fit.

It will be possible after JupyterLab 4.0 release, and we could port it back to LSP extension 4.0 version so that it is available for JupyterLab 3.x. The code in JupyterLab 4.0 is in:

https://github.com/jupyterlab/jupyterlab/blob/8e08e4252a79a6c816535b9e80759adff984cad7/packages/lsp-extension/src/index.ts#L56-L80

and in:

https://github.com/jupyterlab/jupyterlab/blob/a650dd8ac4fefe42d9a3b38a702b8660adf94bc3/packages/lsp/src/extractors/text_extractor.ts

I will look into this during refactor for 4.0.

**Comment by teucer** [comment]
What is the ETA for JupyterLab 4.0?

**Comment by krassowski** [comment]
Unofficially/optimistically end of the year/January 2023.

**Comment by bruttif** [comment]
Following.
